scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -1683,32 +1695,83 @@
         (is (= [foo bar fie] (->> "T3" (get @messages-store) vec)))
         (is (= [foo] (->> "T4" (get @messages-store) vec)))))))
 
+(defn- progress-entry
+  [message percent sequence & {:keys [instance-id]}]
+  (cond-> {:progress-message message
+           :progress-percent percent
+           :progress-sequence sequence}
+          instance-id (assoc :instance-id instance-id)))
+
 (deftest test-progress-aggregator
-  (testing "basic update from inital state"
-    (is (= {"i1" {:progress-message "i1.m1" :progress-percent 10}}
-           (sched/progress-aggregator 10 {} {:instance-id "i1" :progress-message "i1.m1" :progress-percent 10}))))
-
-  (testing "update state for known instance"
-    (is (= {"i1" {:progress-message "i1.m2" :progress-percent 20}}
-           (sched/progress-aggregator 10 {"i1" {:progress-message "i1.m1" :progress-percent 10}}
-                                      {:instance-id "i1" :progress-message "i1.m2" :progress-percent 20}))))
-
-  (testing "handle threshold exceeded"
-    (is (= {"i1" {:progress-message "i1.m1" :progress-percent 10}}
-           (sched/progress-aggregator 1 {"i1" {:progress-message "i1.m1" :progress-percent 10}}
-                                      {:instance-id "i2" :progress-message "i2.m1" :progress-percent 20}))))
-
-  (testing "handle threshold limit reached"
-    (is (= {"i1" {:progress-message "i1.m1" :progress-percent 10}
-            "i2" {:progress-message "i2.m1" :progress-percent 20}}
-           (sched/progress-aggregator 2 {"i1" {:progress-message "i1.m1" :progress-percent 10}}
-                                      {:instance-id "i2" :progress-message "i2.m1" :progress-percent 20})))))
+  (let [pending-progress-threshold 10
+        sequence-cache-store-fn (fn [& {:keys [data threshold] :or {data {} threshold 100}}]
+                                  (atom (cache/lru-cache-factory data :threshold threshold)))]
```

Comment:
Is a TTL-based cache more appropriate?

Reply:
I think it would still need to be backed by a cache of bounded size, e.g. LRU. So I chose to just use LRU directly and bypass the TTL cache requirement. Since the data is just a map and an integer, I was okay skipping the TTL eviction.

Reply:
Added ttl cache.